### DISCUSSION TITLE
is there anyway to split config file into different parts

### DISCUSSION THREAD

**Discussion by ogios** [YaLTeR/niri#275] [Q&A] [answered]
it would be much more easy to read if the config file is too long

**Comment by YaLTeR** [comment]
Not at the moment. Well, I guess you could split and then have a script that cats them all together into the final config.kdl.

**Reply by ogios** [reply, marked as answer]
big brain
haven't thought of this🤯

**Reply by sodiboo** [reply]
> I guess you could split and then have a script that cats them all together into the final config.kdl.

On the topic of concatenating `config.kdl`: This doesn't work. niri doesn't play nicely with defining a "section" multiple times, say e.g. partially in different files.

Minimal example:
```sh
echo "binds; binds" | niri validate -c /proc/self/fd/0
```

I'd quite like to have for my nix module something like `extraConfig` so that users can add additional config options i haven't gotten to yet, instead of either waiting for me to implement them properly or forking my module themselves. Currently i don't, because its usefulness is incredibly limited by this fact. This is in part why my `kdl` module is useful (moreso correctness of serialization), because in theory it allows somewhat structured manipulation of the resulting document. but i don't do anything with this because i didn't come up with any sane way to actually implement such manipulation.

it would be entirely unnecessary if niri could natively handle merging multiply-defined sections

**Reply by YaLTeR** [reply]
Yeah, only splitting into separate sections will work currently, as you mentioned. Which could still be useful for the purpose of splitting a big config file into smaller parts.

**Comment by nebulosa2007** [comment]
Just delete all comments and empty lines from config file and use the Wiki for configuring - the file won't be so long.

**Comment by dantefromhell** [comment]
I have another use-case for split configs.
I'm using @sodiboo niri-nix flake and due to how NixOS works testing a certain parameter requires rebuilding my systems configuration which is a rather hassle that slows-down testing a lot.

With some other programs (kitty, hyprland, pyprland) I've established the pattern of setting up a `~/.config/overrides/$PROGRAM` folder and including that folder as the last piece of my managed config.

Now I have a folder in which I can dump a bunch of configuration options that I can quickly test & iterate on until they are working as expected. E.g this has been pretty helpful with developing more complex window matchings, fine-tuning parameters like input delays and more.

It does require the behaviour of later parsed config files overriding previously configured parameters though to work properly.
